app/tournament/player.js

```diff
@@ -0,0 +1,23 @@
+export class Player {
+    constructor(name, rank) {
+        this.name = name;
+        this.rank = rank;
+    }
+}
+
+export function validatePlayer(data) {
+    var validationResult = {};
+
+    validationResult.error = validateName(data.name) + validateRank(data.rank);
+    validationResult.success = validationResult.error.length === 0;
+
+    return validationResult;
+}
+
+function validateName(name) {
+    return (!name) ? "Name cannot be empty. " : "";
+}
+
+function validateRank(rank) {
+    return (rank < 1) ? "Rank has to be greater than 0." : "";
```

Comment:
My point is that the logic is assuming a string like a number without further parsing. I'd be explicit on that, perhaps out of this validation logic, to ensure that the rank is always a number, not a string representing a number.